src/EnergyPlus/SimulationManager.cc

```diff
@@ -209,6 +209,8 @@ namespace SimulationManager {
              state.dataInputProcessing->inputProcessor->getNumObjectsFound(state, "RunPeriod:CustomRange") > 0 || state.dataSysVars->FullAnnualRun);
         state.dataErrTracking->AskForConnectionsReport = false; // set to false until sizing is finished
 
+        state.init_state(state);
```

Comment:
I decided to plant the call to the uber-function here. Is there a better place for it?

Reply:
It appears this allows calling all getInput functions before the simulation gets started. And yes, order of calls in EnergyPlusData::init_state is critically important. This will replace the isInputObjectUsed function down at line 229 which is also trying to get information prior to the simulation starting. So somewhere between line 212 and 229 seems appropriate to me. I think at the very least that this call should be moved down below OpenOutputFiles at line 214. Or there may be a need to reorganize calls between lines 214 and 229 where some of those lines could be moved to init_state or reorganized locally here depending on what each function does.

Reply:
If I'm following correctly, ideally, we want to split up getInput into two parts: read and cross-check. e.g. Read all of the inputs but don't do any checking at first (e.g. checking for a zone name). Then the input order won't be (or could be less of) an issue.

Reply:
Since getInputs write to output files (err for example) this call should be moved below OpenOutputFiles before merging. getFluidProperties does call Show*Error so the err file needs to be open first.

Reply:
> It appears this allows calling all getInput functions before the simulation gets started. And yes, order of calls in EnergyPlusData::init_state is critically important. This will replace the isInputObjectUsed function down at line 229 which is also trying to get information prior to the simulation starting. So somewhere between line 212 and 229 seems appropriate to me. I think at the very least that this call should be moved down below OpenOutputFiles at line 214. Or there may be a need to reorganize calls between lines 214 and 229 where some of those lines could be moved to init_state or reorganized locally here depending on what each function does.

Okay, I can move this wherever and we can keep moving it if needed. For now, it's essentially an empty function.

Reply:
> If I'm following correctly, ideally, we want to split up getInput into two parts: read and cross-check. e.g. Read all of the inputs but don't do any checking at first (e.g. checking for a zone name). Then the input order won't be (or could be less of) an issue.

You're thinking five steps ahead here. I'm really just trying to peel off objects that themselves don't reference any other objects at this point. But, yes, eventually this will be a "read" pass followed by a "link" pass. And I believe the "read" pass may be able to take place automatically, based on some stuff that the ORNL boys are working on.

Reply:
> Since getInputs write to output files (err for example) this call should be moved below OpenOutputFiles before merging. getFluidProperties does call Show*Error so the err file needs to be open first.

Did you already do this?

Reply:
No, but it does need to move. I'm not sure about the other calls below but that will show itself, as well as order dependence, once things start moving to the new init_state function.

Reply:
OK, I just checked. OpenOutputFiles opens eso, eio, mtr, and bnd. I think the bnd (node list) needs to be open before calling getInput. The err file gets opened in initializeEnergyPlus and initializeAsLibrary which seems it's already open by this point. So this line still needs to move after OpenOutputFiles.

Reply:
I moved it, although it looked like it was already moved for some reason (that's why I thought you did it). I'm going to make this transition over time, this just lets me get started.

Reply:
Regarding: "split up getInput into two parts: read and cross-check" I think if we want to read inputs early we would want to know that data is valid and error checked so that there is no need for other checks elsewhere (i.e., if zoneNum > 0), etc.). Just suggesting this does not move too fast other than trying to get a good order for getInput calls. Once that effort is well under way we could make decisions like that.